### DISCUSSION TITLE
Can anyone help me ?

### DISCUSSION THREAD

**Discussion by anshumanthakur24** [Cyfrin/foundry-full-course-cu#2198] [General]
`.zksync/zksolc-linux-amd64-musl-v1.5.1": Permission denied (os error 13) `

I am getting error like this while using 

`forge build --zksync` command. 

I am using WSL on windows 11.

Everything works fine except zksync. Tried reinstalling it too. Any suggestions??

I looked for the answer but couldn't find anything that worked ( probably cuz i am rookie) but any help will be very much appreciated.

**Comment by EngrPips** [comment]
Hello, I think this has to do with logging in the user that has administrative access on your command line. Could you check what access the current user in your command line has? You can open the command line as an administrator, and I think it should help avoid the `Permission denied` error. You can learn how to open the command line as administrator here https://grok.lsu.edu/article.aspx?articleid=18026&printable=y

**Reply by anshumanthakur24** [reply]
I am using WSL terminal inside the VS Code. How could I check that what access the current user have in that?

**Reply by EngrPips** [reply]
This https://hcc.unl.edu/docs/handling_data/data_storage/linux_file_permissions/#:~:text=Type%20the%20command%20ls%20%2Dl,is%20shown%2C%20it's%20a%20file. article should help with understanding how to check the access of the current users.

**Reply by anshumanthakur24** [reply]
i did run the `ls -1` command and it showed 2 folders

simpleStorage ( the one i am working on)
foundry-zksync

I looked around and came to conclusion that i should run this command

`chmod +x /home/anshuman/.zksync/zksolc-linux-amd64-musl-v1.5.1`

to maybe give it the necessary permissions. Should i do it? cuz i am worried it might mess up things.

**Reply by EngrPips** [reply]
Honestly, I don't know much to be confident to tell you to do it but the worst case is it messes up things and you can start again.

**Reply by EngrPips** [reply]
Do you try to read about what this command `chmod +x /home/anshuman/.zksync/zksolc-linux-amd64-musl-v1.5.1` does?

**Reply by anshumanthakur24** [reply]
It says 

This command adds execute permissions to the file.

**Reply by EngrPips** [reply]
If that is what it does I think it makes sense to run it

**Reply by anshumanthakur24** [reply]
crap I got a different error now 

`"/home/anshuman/.zksync/zksolc-linux-amd64-musl-v1.5.1": Exec format error (os error 8)`

**Reply by EngrPips** [reply]
Different error usually mean progress, i hope the same for this case lol.

**Reply by EngrPips** [reply]
These are some similar discussions that you can check for solution.

https://github.com/Cyfrin/foundry-full-course-cu/discussions/1667

https://github.com/Cyfrin/foundry-full-course-cu/discussions/808

**Reply by Maa-ly** [reply]
You're getting a permission because your user doesn't have the necessary permissions to execute the file.
run this "chmod +x ~/.zksync/zksolc-linux-amd64-musl-v1.5.1" 
And make sure the file is  owned by the user you're logged in as

**Reply by EngrPips** [reply]
Hello @anshumanthakur24, Did you resolve this already?

**Reply by anshumanthakur24** [reply]
Umm how do I say it.

 Here is what i did.

I reinstalled foundry and then zksync foundry. Still didn't worked.

Then I headed to 

`.zksync/zksolc-linux-amd64-musl-v1.5.1`

in file explorer. The file there was sus! it was 0KB so I get a urge to delete it and followed my gut feeling and it works out to be fine. 

I don,t know how but it worked. LOL

**Reply by anshumanthakur24** [reply]
Thanks @EngrPips and @Maa-ly for providing me assistance. Very appreciated. I certainly learned something from this conversation. Thanks mates.

**Reply by EngrPips** [reply]
You are welcome, Friend. Keep crushing it.
